### PR TITLE
Adding support for startd checks

### LINKED ISSUE #104: monitoring for glidein startup (before the startd connects)
We should monitor the glidein startup, before the startd connects.

### PR DESCRIPTION
**Pull request by hskarlupka**
Pyglidein now sets scripts for checking GPUs, CVMFS, and GridFTP services on each node.  The scripts are hooked into condor startd cron service and produce classads based on the success or failure of the check script. The GPU scrip also outputs a metrics classad for measuring the performance of GPUs. Resolve #104

### CODE REVIEW

File: pyglidein/startd_cron_scripts/clsim_gpu_test.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    parser = OptionParser(usage)
+    parser.add_option('-n', type='int', default=1,
+                      help="Number of simulations to run")
+    (options, args) = parser.parse_args()
+
+    try:
+        # This line is needed to access the nvidia gpus via OpenCL inside a container.
+        # TODO: Is this needed in a grid environment?
+        os.environ['LD_LIBRARY_PATH'] = ('/usr/local/nvidia/lib64:'
+                                         '{}'.format(os.environ['LD_LIBRARY_PATH']))
```
Comment (dsschult) [inline pyglidein/startd_cron_scripts/clsim_gpu_test.py]: `os.environ[]` should already return a string.  is there a reason to use `.format()` here, or would something like this be better?
```python
        os.environ['LD_LIBRARY_PATH'] = ('/usr/local/nvidia/lib64:' + os.environ['LD_LIBRARY_PATH'])
```

Also, I should note that we might need to check that the key exists in `os.environ` before blindly using.

File: pyglidein/startd_cron_scripts/clsim_gpu_test.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+        cmd = '{} {}'.format(os.path.join('/cvmfs/icecube.opensciencegrid.org/py2-v2',
+                                          os.environ['OS_ARCH'],
+                                          'metaprojects/simulation/V05-00-07/env-shell.sh'),
+                             os.path.join('/cvmfs/icecube.opensciencegrid.org/py2-v2',
+                                          os.environ['OS_ARCH'],
+                                          'metaprojects/simulation/V05-00-07/clsim/resources',
+                                          'scripts/benchmark.py'))
+        cmd += ' -n {}'.format(options.n)
+        check_output(cmd, shell=True, env=os.environ, stderr=STDOUT)
```
Comment (dsschult) [inline pyglidein/startd_cron_scripts/clsim_gpu_test.py]: Is there a reason why we do `shell=True`?  That's potentially a security hole, and also maybe unnecessary.

Also, why `env=os.environ`?  That's the default anyway.

File: pyglidein/startd_cron_scripts/clsim_gpu_test.py (hunk 1 above)
Comment (hskarlupka) [reply pyglidein/startd_cron_scripts/clsim_gpu_test.py]: Fixed.

File: pyglidein/startd_cron_scripts/clsim_gpu_test.py (hunk 2 above)
Comment (hskarlupka) [reply pyglidein/startd_cron_scripts/clsim_gpu_test.py]: Good call.  I switched to `shell=False` and cleaned up the commands.

**Pull request merged**